chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -346,31 +346,71 @@
   mode "0644"
 end
 
+is_admin = CrowbarHelper.is_admin?(node)
 crowbar_node = node_search_with_cache("roles:crowbar").first
 address = crowbar_node["crowbar"]["network"]["admin"]["address"]
 protocol = crowbar_node["crowbar"]["apache"]["ssl"] ? "https" : "http"
 server = "#{protocol}://#{address}"
-is_admin = CrowbarHelper.is_admin?(node)
-if is_admin
-  username = "crowbar"
-  password = crowbar_node["crowbar"]["users"][username]["password"]
-else
-  username = crowbar_node["crowbar"]["client_user"]["username"]
-  password = crowbar_node["crowbar"]["client_user"]["password"]
-end
 verify_ssl = !crowbar_node["crowbar"]["apache"]["insecure"]
 
 package "ruby2.1-rubygem-crowbar-client"
 
-template "/etc/crowbarrc" do
-  source "crowbarrc.erb"
-  variables(
-    server: server,
-    username: username,
-    password: password,
-    verify_ssl: verify_ssl
-  )
-  owner "root"
-  group "root"
-  mode "0o600"
+if is_admin && ::File.exist?("/etc/crowbarrc")
+  # On admin server, only make sure the address and verify_ssl options are
+  # correct; the admin is the one controlling the username & password
+
+  # After installation of a gem, we have a new path for the new gem, so we
+  # need to reset the paths if we can't load the gem
+  begin
+    require "inifile"
+  rescue LoadError
+    Gem.clear_paths
+  end
+
+  begin
+    crowbarrc = IniFile.load("/etc/crowbarrc")
+
+    crowbarrc_config = crowbarrc["default"]
+
+    if server != crowbarrc_config["server"]
+      crowbarrc_config["server"] = server
+      Chef::Log.info("Will update \"server\" option in /etc/crowbarrc to \"#{server}\"")
+      do_save = true
+    end
+
+    crowbarrc_verify_ssl = crowbarrc_config["verify_ssl"].nil? ||
+      ![false, 0, "0", "f", "F", "false", "FALSE"].include?(crowbarrc_config["verify_ssl"])
+
+    if protocol == "http" && !crowbarrc_config["verify_ssl"].nil?
+      crowbarrc_config.delete("verify_ssl")
+      Chef::Log.info("Will remove \"verify_ssl\" option in /etc/crowbarrc")
+      do_save = true
+    elsif protocol == "https" && verify_ssl != crowbarrc_verify_ssl
+      crowbarrc_config["verify_ssl"] = verify_ssl ? 1 : 0
+      Chef::Log.info("Will update \"verify_ssl\" option in /etc/crowbarrc to " \
+          "\"#{crowbarrc_config["verify_ssl"]}\"")
```

Comment:
Is there a reason why you choose to use 0/1 as default values instead of just the boolean value of `verify_ssl`?

Reply:
That's what we did in the past (see the old code that is just moved)